src/category_theory/sites/dense_subsite.lean

```diff
@@ -0,0 +1,415 @@
+  ℱ.val.obj (op X) ≅ ℱ'.val.obj (op X) :=
+{ hom := app_hom H i.hom X,
+  inv := app_hom H i.inv X,
+  hom_inv_id' := by { ext x, apply H.ext, intros Y f, simp },
+  inv_hom_id' := by { ext x, apply H.ext, intros Y f, simp } }
+
+/--
+Given an natural transformation `G ⋙ ℱ ⟶ G ⋙ ℱ'` between presheaves of types, where `G` is full
+and cover-dense, and `ℱ'` is a sheaf, we may obtain a natural transformation between sheaves.
+-/
+@[simps] noncomputable
+def presheaf_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) : ℱ ⟶ ℱ'.val :=
+{ app := λ X, app_hom H α (unop X), naturality' := λ X Y f,
+  begin
+    ext x,
+    apply H.ext ℱ' (unop Y),
+    intros Y' f',
+    simp only [app_hom_restrict, types_comp_apply, ← functor_to_types.map_comp_apply],
+    rw app_hom_restrict H α (f ≫ f'.op : op (unop X) ⟶ _)
+  end }
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of types, where `G` is full and
+cover-dense, and `ℱ, ℱ'` are sheaves, we may obtain a natural isomorphism between presheaves.
+-/
+@[simps] noncomputable
+def presheaf_iso {ℱ ℱ' : SheafOfTypes.{v} K} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) :
+  ℱ.val ≅ ℱ'.val :=
+nat_iso.of_components (λ X, app_iso H i (unop X)) (presheaf_hom H i.hom).naturality
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of types, where `G` is full and
+cover-dense, and `ℱ, ℱ'` are sheaves, we may obtain a natural isomorphism between sheaves.
+-/
+@[simps] noncomputable
+def sheaf_iso {ℱ ℱ' : SheafOfTypes.{v} K} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) : ℱ ≅ ℱ' :=
+{ hom := (presheaf_iso H i).hom, inv := (presheaf_iso H i).inv,
+  hom_inv_id' := (presheaf_iso H i).hom_inv_id, inv_hom_id' := (presheaf_iso H i).inv_hom_id }
+
+
+end types
+open types
+
+variables {ℱ : Dᵒᵖ ⥤ A} {ℱ' : Sheaf K A}
+
+/-- (Implementation). The sheaf map given in `types.sheaf_hom` is natural in terms of `X`. -/
+@[simps] noncomputable
+def sheaf_coyoneda_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  coyoneda ⋙ (whiskering_left Dᵒᵖ A Type*).obj ℱ ⟶
+  coyoneda ⋙ (whiskering_left Dᵒᵖ A Type*).obj ℱ'.val :=
+{ app := λ X, presheaf_hom H (hom_over α (unop X)), naturality' := λ X Y f,
+  begin
+  ext U x,
+  change app_hom H (hom_over α (unop Y)) (unop U) (f.unop ≫ x) =
+    f.unop ≫ app_hom H (hom_over α (unop X)) (unop U) x,
+  symmetry,
+  apply sheaf_eq_amalgamation,
+  apply H.is_cover,
+  intros Y' f' hf',
+  change unop X ⟶ ℱ.obj (op (unop _)) at x,
+  simp only [pushforward_family, functor.comp_map, coyoneda_obj_map, hom_over_app, category.assoc],
+  congr' 1,
+  conv_lhs { rw ← hf'.some.fac },
+  simp only [← category.assoc, op_comp, functor.map_comp],
+  congr' 1,
+  refine (app_hom_restrict H (hom_over α (unop X)) hf'.some.map.op x).trans _,
+  simp
+  end }
+
+/--
+(Implementation). `sheaf_coyoneda_hom` but the order of the arguments of the functor are swapped.
+-/
+@[simps] noncomputable
+def sheaf_yoneda_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  ℱ ⋙ yoneda ⟶ ℱ'.val ⋙ yoneda :=
+begin
+  let α := sheaf_coyoneda_hom H α,
```

Comment:
What does `simps` do with the let statement? Does the output of `simps?` look the way you want it to be?

Reply:
It actually gives quite a decent looking lemma.
But now that I think about it, I probably want to make it the implementation details more opaque, so I removed them.